CMakeLists.txt

```diff
@@ -45,40 +30,88 @@ file(
+ELSEIF (UNIX)
+    message("🐧 Building for Linux")
+    set(BINARY_SUFFIX "")
+ELSE ()
+    message(FATAL_ERROR "❌ Unsupported host system: ${CMAKE_HOST_SYSTEM_NAME}")
+ENDIF ()
+
 # ======== Python Checks ========
 IF (NO_VENV)
-    IF ("${CMAKE_HOST_SYSTEM_NAME}" STREQUAL "Windows")
+    IF (WIN32)
         set(PIP_COMMAND pip)
         set(PYTHON_COMMAND python)
     ELSE ()
         set(PIP_COMMAND pip3)
         set(PYTHON_COMMAND python3)
     ENDIF ()
 ELSE ()
-    IF ("${CMAKE_HOST_SYSTEM_NAME}" STREQUAL "Windows")
+    IF (WIN32)
         set(PYTHON_COMMAND pipenv run python)
     ELSE ()
         set(PYTHON_COMMAND pipenv run python3)
     ENDIF ()
 ENDIF ()
 
 find_package(Python3 COMPONENTS Interpreter)
-message("Python Command Set to ${PYTHON_COMMAND}")
 find_program(PYTHON_HAS_COMMAND ${PYTHON_COMMAND})
 IF (Python3_VERSION VERSION_LESS 3.8)
     message(FATAL_ERROR "❌ Python version 3.8 or greater is required")
 ELSEIF (Python3_VERSION VERSION_GREATER_EQUAL 3.12)
-    message(WARNING "⚠️ Python version 3.12 or greater is not supported as pyOCD does not support it")
+    message(FATAL_ERROR "⚠️ Python version 3.12 or greater is not supported as pyOCD does not support it")
 ENDIF ()
-message("🐍 Found Python ${Python3_VERSION}")
+message("🐍 Found Python ${Python3_VERSION}, ${PYTHON_COMMAND}")
 
 # ======== Python Checks ========
 
-IF ("${PLATFORM}" STREQUAL "x86" OR "${PLATFORM}" STREQUAL "arm")
+message("⚙️ Configuring for ${PLATFORM}-${TARGET} build target")
+IF ("${PLATFORM}" STREQUAL "firmware")
+    # ======== Dependency Handling ========
+    IF (HANDLE_DEPS)
+        include(firmware/cmake/dependencies.cmake)
+    ENDIF()
+
+    # ======== Compiler ========
+    IF("${TARGET}" STREQUAL "arm")
+        # On Apple silicon, this line mysteriously causes loading CMake for unit tests to fail.
+        # However, it is equally mysteriously required for building embedded binaries.
+        # So, only run this if loading the embedded configuration.
+        # TODO: Look into what is causing this.
+        set(CMAKE_C_COMPILER_WORKS 1)
+        set(TOOLCHAIN_PREFIX ${TOOLCHAIN_PREFIX}arm-none-eabi)
+        set(CMAKE_C_COMPILER ${TOOLCHAIN_PREFIX}-gcc${BINARY_SUFFIX})
+        set(CMAKE_CXX_COMPILER ${TOOLCHAIN_PREFIX}-g++${BINARY_SUFFIX})
+        set(CMAKE_ASM_COMPILER ${TOOLCHAIN_PREFIX}-gcc${BINARY_SUFFIX})
+        set(CMAKE_OBJCOPY ${TOOLCHAIN_PREFIX}-objcopy${BINARY_SUFFIX})
+        set(CMAKE_OBJDUMP ${TOOLCHAIN_PREFIX}-objdump${BINARY_SUFFIX})
+        set(SIZE ${TOOLCHAIN_PREFIX}-size${BINARY_SUFFIX})
+        message("✅ ARM Toolchain Included, using ${CMAKE_C_COMPILER}")
+        set(CMAKE_SYSTEM_NAME Generic)
+        project("UBC Formula Electric Software" C ASM)
+    ELSEIF ("${TARGET}" STREQUAL "x86")
+        project("UBC Formula Electric Software" C CXX)
+    ELSE()
+        message(FATAL_ERROR "❌ Please set the TARGET variable to 'arm' or 'x86'. To do this, set the -DTARGET=<target>")
+    ENDIF()
```

Comment:
Sorry but on second thought do you mind changing this back to PLATFORM=firmware and TARGET=deploy/test? It'd be nice to keep this unified across the whole repo (as you suggested) :)
Maybe name the build directories `build_fw_deploy` and `build_fw_test` to keep them concise.